CONTRIBUTING.md

```diff
@@ -1,47 +1,205 @@
+
+- Commit message should describe motivation. Think about your code reviewers and what information they need in order to understand what you did. If it's a big commit (hopefully not), try to provide some good entry points so it will be easier to follow.
+
+### Step 6: Pull Request
+
+- Push your changes to your GitHub fork
+- Submit a Pull Requests on the aws-sdk-ios repo to the `master` branch and add one of the amplify-ios team members on it (@kneekey23, @lawmicha, @wooj2, @palpatim, @royjit, @drochetti, @phani-srikar).
+- The title of your PR must be descriptive to the specific change.
+- No period at the end of the title.
+- Pull Request message should indicate which issues are fixed: `fixes #<issue>` or `closes #<issue>`.
+- PR messaged should include shout out to collaborators.
+- If not obvious (i.e. from unit tests), describe how you verified that your change works.
+- If this PR includes breaking changes, they must be listed at the top of the changelog as described above in the Pull Request Checklist.
+- Discuss review comments and iterate until you get at least one “Approve”. When iterating, push new commits to the same branch. 
+- Usually all these are going to be squashed when you merge to master.
+- Make sure to update the PR title/description if things change. 
+- Rebase with master if the master branch has commits ahead of your fork.
+
+### Step 7: Merge
+Once your PR has been approved and tested, go ahead and merge it into `master`! Barring any unforeseen circumstances, your changes will be released in our next version. Yay!! 
+
+## Troubleshooting
+
+Some build issues can be solved by [removing your derived data](https://iosdevcenters.blogspot.com/2015/12/how-to-delete-derived-data-and-clean.html) and doing a clean and build. For any other serious build issues, open a new issue or see if there is one existing that may have a fix on it.
+
+## Related Repositories
+
+We like to have platform parity as much as possible while still respecting the language we are writing in so often times we will try to align our designs with the [amplify-android](https://github.com/aws-amplify/amplify-android) repository. Feel free to check it out as well to see how something is done for Android.
 
 ## Finding contributions to work on
-Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any ['help wanted'](https://github.com/awslabs/amplify-ios/labels/help%20wanted) issues is a great place to start.
+Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any ['help wanted'](https://github.com/aws-amplify/amplify-ios/labels/help%20wanted) issues is a great place to start.
```

Comment:
can we add "good first issue" link as well?

Reply:
added